resourcemanager/pool/spool/spool.go

```diff
@@ -173,6 +181,10 @@ func (p *Pool) checkAndAddRunningInternal(concurrency int32) (conc int32, run bo
 // ReleaseAndWait releases the pool and waits for all tasks to be completed.
 func (p *Pool) ReleaseAndWait() {
 	p.isStop.Store(true)
+	// wait for all the task in the pending to exit
+	for p.waiting.Load() > 0 {
+		time.Sleep(waitInterval)
```

Comment:
I want to confirm the usage of `ReleaseAndWait`. If it may be called in `Executor.Close`, `5ms` sleep is unacceptable. If it's called when TiDB exiting or kvstore closing, this LGTM.

Reply:
I remove ```5ms``` sleep first.

Reply:
A spinlock is risky because it may loop here while the `Run` or `RunWithConcurrency` threads are scheduled out, why not use a condition variable here?

Reply:
Done